public/js/index.js

```diff
@@ -24,10 +24,25 @@ const diagram = $(go.Diagram, "diagramDiv",
         "toolManager.hoverDelay": 100
     });
 
+function showMessage(s) {
+    document.getElementById("diagramEventsMsg").textContent = s;
+  }
+
+diagram.addDiagramListener("ObjectSingleClicked",
+    function(e) {
+        let part = e.subject.part;
+        let str = "ID: " + part.data.key + "\nName: " + part.data.nameProperty + "\nCategory: "
+            + part.data.category + "\nDescription: " + part.data.desc + "\nTags: "
+            + part.data.tags + "\nVersion: " + part.data.version + "\nDepartment: "
+            + part.data.department + "\nAllowed Users: " + part.data.allowedUsers
+            + "\nLicense: " + part.data.license + "\nLocation: " + part.data.loc;
+        if (!(part instanceof go.Link)) showMessage(str);
+    });
+
```

Comment:
It would be nice to have comments here, so everyone knows what the function is used for. 
Somhting like this: 
```javascript
/**
 * What the function does
 * param s 'What is s'
*/
function showMsg(s){...}
```